poc2/public/script/wordlehelper.js

```diff
@@ -393,21 +393,15 @@ $(document).ready(function(){
       */
       if (arrNotPresentLetters.length > 0)
       {
-      console.log("Removing present letters from non-present letters")
+      console.log("Removing present and correct letters from non-present letters")
         //remove present letters from the not-present letters
         arrPresentLettersSet = new Set(arrPresentLetters);
+        arrCorrectLettersSet = new Set(arrCorrectLetters);
+        arrPresentAndCorrectLettersSet = new Set([...arrPresentLettersSet, ...arrCorrectLettersSet]);
         arrNotPresentLetters = arrNotPresentLetters.filter( (x) => {
-          return !arrPresentLettersSet.has(x);
+          return !arrPresentAndCorrectLettersSet.has(x);
```

Comment:
i would probably define this as something like
 return !(arrPresentLettersSet.has(x) || arrPresentAndCorrectLettersSet.has(x))